### PR TITLE
enhance: build word index prune word in comments

### LINKED ISSUE #894: [Enhancement] find reference support pruning on word index
## Enhancement

prune on word index

- [ ] exclude words in doc strings
- [ ] exclude words in comments

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 

fix #894 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

enhance: build word index prune word in comments

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

**Review by Peefy — APPROVED** [review]
LGTM

File: kclvm/tools/src/LSP/src/find_refs.rs [hunk 1]
```diff
@@ -126,7 +126,7 @@ mod tests {
     fn setup_word_index_map(root: &str) -> HashMap<Url, HashMap<String, Vec<Location>>> {
         HashMap::from([(
             Url::from_file_path(root).unwrap(),
-            build_word_index(root.to_string()).unwrap(),
+            build_word_index(root.to_string(), true).unwrap(),
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/find_refs.rs:129]: Format check CI failed.

Comment (amyXia1994) [reply kclvm/tools/src/LSP/src/find_refs.rs:129]: fixed

Comment (Peefy) [reply kclvm/tools/src/LSP/src/find_refs.rs:129]: > fixed

Unit test CI failed.

**Pull request merged**